### ISSUE TITLE
Idea: Self brokered statuspage-like system for tenants

### ISSUE THREAD

**Issue by jcscottiii** [18F/cg-agent-q#4]
![image](https://cloud.githubusercontent.com/assets/7788930/17623755/a946f850-606f-11e6-93bb-419c24a50226.png)

https://sourcegraph.github.io/checkup/

Sourcegraph created `Checkup` which is for "Simple uptime monitoring distributed, self-hosted health checks and status pages"
## Current Status of Checkup:
- There are three components to Checkup:
  1. The status page is just a static site
     - Could just push a static site with a route of the tenant's choosing upon creation.
  2. The `checkup` command / daemon (which can also be used as a Go library).
     - The broker could use the library part and manage all the periodic checks for all the site
     - Concern/ Question: will the broker break after trying to manage X cron-like schedules?
  3. Storage: Only supports S3
## More details:
- It currently only uses S3 for storage of data.
  - Question: Do we want to use that? If not, what kind of storage would we use.
- It currently uses two sets of S3 credentials. 1) A set of permissions that can write to the S3 bucket [which is backend code](https://github.com/sourcegraph/checkup/blob/9a91b6ea2cfac915abb4c1a352ea79fdad4e8831/s3.go#L37) and 2) A set of permissions that can read from the bucket that [are public in the frontend code](https://github.com/sourcegraph/checkup/blob/master/statuspage/js/config_template.js#L14)
  - Opinion: Depending on how you scope s3 permissions, you would have to also manage a lot of permissions (e.g. a new pair of read and write AWS creds for each brokered status page)
## Extra work to cross the finish line
- There are no implemented notifiers right but there is an interface `Notifier` that can developers can implement and the library will kick off any registered `Notifier`s. 
  - Would we want to do e-mail, slack, etc?
- The broker API can support misc json but from a UX perspective it could easily be messed up
  - Maybe create a plugin that can wrap all of that creation information?

**Comment by afeld** [comment]
One consideration is that you generally don't want status pages to be running on the same infrastructure as the system it's monitoring, so the broker would need to deploy it to AWS (or somewhere) outside of cloud.gov.

**Comment by jcscottiii** [comment]
Yeah that's true. Monitoring is useless if the system co-hosting your app and the monitor goes down lol.
In terms of expanding, we could deploy it at the BOSH layer (some of the brokers are deployed at this level such as the docker-swarm broker, and our monitoring with Riemann is here too). Thoughts?

**Comment by afeld** [comment]
The separate-er the better. Some sites like Travis CI will even have the status page at a separate domain in case there are domain registration or DNS issues with their canonical one.

https://www.traviscistatus.com/

**Comment by mogul** [comment]
If this is for app-hosting, it's fine to deploy it on cloud.gov, because it is next to the thing its monitoring, not dependent on it.
